src/BackgroundTask/MonitorCssTransientCaching.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * Class MonitorCssTransientCaching.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\BackgroundTask;
+
+use AMP_Options_Manager;
+use AmpProject\AmpWP\Option;
+use DateTimeImmutable;
+use DateTimeInterface;
+use Exception;
+
+/**
+ * Monitor the CSS transient caching to detect and remedy issues.
+ *
+ * This checks whether there's excessive cycling of CSS cached stylesheets and disables transient caching if so.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class MonitorCssTransientCaching extends CronBasedBackgroundTask {
+
+	/**
+	 * Name of the event to schedule.
+	 *
+	 * @var string
+	 */
+	const EVENT_NAME = 'amp_monitor_css_transient_caching';
+
+	/**
+	 * Key to use to persist the time series in the WordPress options table.
+	 *
+	 * @var string
+	 */
+	const TIME_SERIES_OPTION_KEY = 'amp_css_transient_monitor_time_series';
+
+	/**
+	 * Default threshold to use for problem detection in number of transients per day.
+	 *
+	 * @var float
+	 */
+	const DEFAULT_THRESHOLD = 50.0;
+
+	/**
+	 * Sampling range in days to calculate the moving average from.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_SAMPLING_RANGE = 14;
+
+	/**
+	 * Get the interval to use for the event.
+	 *
+	 * @return string An existing interval name. Valid values are 'hourly', 'twicedaily' or 'daily'.
+	 */
+	protected function get_interval() {
+		return self::DEFAULT_INTERVAL_DAILY;
+	}
+
+	/**
+	 * Get the event name.
+	 *
+	 * This is the "slug" of the event, not the display name.
+	 *
+	 * Note: the event name should be prefixed to prevent naming collisions.
+	 *
+	 * @return string Name of the event.
+	 */
+	protected function get_event_name() {
+		return self::EVENT_NAME;
+	}
+
+	/**
+	 * Process a single cron tick.
+	 *
+	 * @todo This has arbitrary arguments to allow for testing, as we don't have dependency injection for services.
+	 *       With dependency injection, we could for example inject a Clock object and mock it for testing.
+	 *
+	 * @param DateTimeInterface $date            Optional. Date to use for timestamping the processing (for testing).
+	 * @param int               $transient_count Optional. Count of transients to use for the processing (for testing).
+	 * @return void
+	 * @throws Exception If a date could not be instantiated.
+	 */
+	public function process( DateTimeInterface $date = null, $transient_count = null ) {
+		if ( $this->is_css_transient_caching_disabled() ) {
```

Comment:
The monitor should short-circuit if an external object cache is used, no? The query in `query_css_transient_count` will always be 0:

```suggestion
		if ( wp_using_ext_object_cache() || $this->is_css_transient_caching_disabled() ) {
```